andela/ah-kgl-avengers-frontend
# #166978152 user can logout

### What does this PR do?
Add logout feature and fix some issues

### Description of Task to be completed?
- Clear the token from  local storage when the user logs out
- Call backend logout API endpoint with the token
- Update the views
- Add tests for the feature and the existing code
- Fix view an article with 0 rates
- Make _my-articles_ and _new-post_ pages accessible to logged in users only
- Remove tags in article editor title when doing copy/paste 
- Can update article tags and title

### How should this be manually tested?
```sh
git pull origin
git checkout ft-user-logout-166978152
npm install
npm start    // will open the browser
```

- Navigate to `localhost:8000/login` to login
- On any page you are on, use the dropdown menu (which comes by clicking the user avatar) and click on logout

- Repeat the same steps, but log out from multiple pages to see the redirects

Run local tests:
```sh
git pull origin
git checkout ft-user-logout-166978152
npm install
npm test
```

### Any background context you want to provide?
You need to be logged in 

### What are the relevant pivotal tracker stories?
[Users should log out #166978152](https://www.pivotaltracker.com/n/projects/2321840/stories/166978152)

### Screenshots 
None

### Questions:
None
